reana_client/cli/test.py

```diff
@@ -0,0 +1,174 @@
+
+    def get_workflow_disk_usage(self, workflow, parameters, access_token):
+        """Display disk usage workflow."""
+        return get_workflow_disk_usage(workflow, parameters, access_token)
+
+    def get_workflow_logs(
+        self, workflow, access_token, steps=None, page=None, size=None
+    ):
+        """Get logs from a workflow engine, use existing API function."""
+        return get_workflow_logs(workflow, access_token, steps, page, size)
+
+    def get_workflow_status(self, workflow, access_token):
+        """Get of a previously created workflow."""
+        return get_workflow_status(workflow, access_token)
+
+    def get_workflow_specification(self, workflow, access_token):
+        """Get specification of previously created workflow."""
+        return get_workflow_specification(workflow, access_token)
+
+    def download_file(self, workflow, file_path, access_token):
+        """Download the requested file if it exists."""
+        return download_file(workflow, file_path, access_token)
+
+
+@click.group(help="Workflow run test commands")
+def test_group():
+    """Workflow run test commands."""
+
+
+@test_group.command("test")
+@click.option(
+    "-w",
+    "--workflow",
+    default="workflow",
+    help="Name of workflow to be tested, default is 'workflow'",
+)
+@click.option(
+    "-n",
+    "--test_file",
+    default=None,
+    help="Gherkin file for testing properties of a workflow execution. Overrides files in reana.yaml if provided.",
+)
+@click.option(
+    "-f",
+    "--file",
+    type=click.Path(exists=True, resolve_path=True),
+    default=get_reana_yaml_file_path,
+    help="REANA specification file describing the workflow to "
+    "execute. [default=reana.yaml]",
+)
+@click.option(
+    "--skip-validation",
+    is_flag=True,
+    help="If set, specifications file is not validated before "
+    "submitting its contents to the REANA server.",
+)
+@click.pass_context
+@add_access_token_options
+@check_connection
+def test(ctx, workflow, test_file, file, skip_validation, access_token):
+    r"""
+    Test workflow execution, based on a given Gherkin file.
+
+    Multiple files can be specified in the reana specification file (reana.yaml).
+
+    The ``test`` command allows for testing of a workflow execution,
+    by assessing whether it meets certain properties specified in a
+    chosen feature file.
+
+    Example:
+        $ reana-client test -w myanalysis -n test_analysis.feature
+        $ reana-client test -w myanalysis
+    """
+
+    specification_filename = click.format_filename(file)
+    try:
+        reana_specification = load_validate_reana_spec(
+            specification_filename,
+            access_token=access_token,
+            skip_validation=skip_validation,
+            server_capabilities=True,
+        )
+    except Exception:
+        display_message(f"Error loading {file} specification file.", msg_type="error")
+        sys.exit(1)
```

Comment:
I'm wondering whether we should fetch the specification of the workflow passed with `--workflow` directly from the server, instead of reading it from disk?

Reply:
In this case, we would also not need the `--skip-validation` flag.